### PR TITLE
Extend save parameters to handle multiple realizations

### PR DESCRIPTION
**Pull request by dafeda**
Add test that uses the new functionality and also documents some troublesome behavior of adaptive localization.

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/ert/ui_tests/cli/test_field_parameter.py [hunk 1]
```diff
@@ -210,3 +217,112 @@ def test_parameter_update_with_inactive_cells_xtgeo_grdecl(tmpdir):
             assert "nan" not in Path(
                 "simulations/realization-0/iter-1/my_param.grdecl"
             ).read_text(encoding="utf-8")
+
+
+def test_field_param_update_using_heat_equation__zero_var_params_and_adaptive_loc(
```
[the remark this answers is not in the document]
Comment (dafeda) [reply tests/ert/ui_tests/cli/test_field_parameter.py:224]: Fixed, thanks.

**Comment by xjules** [comment]
Wondering if you don't want to update the existing usages as well:
https://github.com/equinor/ert/blob/5b96e8a9a4cfcd809941596212ea637dcd493018/src/ert/enkf_main.py#L192
https://github.com/equinor/ert/blob/d59ed13cf07d2e547d0a7be2399c2583510dae4d/src/ert/analysis/_es_update.py#L131

There are more ....

**Comment by dafeda** [comment]
> Wondering if you don't want to update the existing usages as well:
> 
> https://github.com/equinor/ert/blob/5b96e8a9a4cfcd809941596212ea637dcd493018/src/ert/enkf_main.py#L192
> 
> 
> https://github.com/equinor/ert/blob/d59ed13cf07d2e547d0a7be2399c2583510dae4d/src/ert/analysis/_es_update.py#L131
> 
> There are more ....

Could perhaps look into it, but I don't think it makes sense in the two examples you've given.
The first one (line 192) calls `sample_or_load` for each realization before calling `ensemble.save_parameters`.
The second one (line 131) calls the `save_parameters` method in `parameter_config` and not ensemble.

File: src/ert/analysis/_es_update.py [hunk 2]
```diff
@@ -505,9 +505,8 @@ def _copy_unupdated_parameters(
 
     # Copy the non-updated parameter groups from source to target for each active realization
     for parameter_group in not_updated_parameter_groups:
-        for realization in iens_active_index:
-            ds = source_ensemble.load_parameters(parameter_group, int(realization))
-            target_ensemble.save_parameters(parameter_group, realization, ds)
+        ds = source_ensemble.load_parameters(parameter_group, iens_active_index)
+        target_ensemble.save_parameters(parameter_group, iens_active_index, ds)
```
Comment (dafeda) [inline src/ert/analysis/_es_update.py:509]: I've used it here though @xjules which I think makes sense 👍

**Pull request merged**